packages/typescript/package.json

```diff
@@ -58,8 +58,8 @@
     "eslint-import-resolver-typescript": "^3.6.3",
     "eslint-plugin-import-x": "^4.3.0",
     "eslint-plugin-jsdoc": "^50.2.4",
-    "typescript": ">=4.8.4 <5.6",
-    "typescript-eslint": "^8.24.0"
+    "typescript": ">=4.8.4 <5.9.0",
+    "typescript-eslint": "^8.24"
```

Comment:
I figured that we could give consumers more flexibility for this package given that sometimes new versions bump the peer dependency on `typescript` and some don't.